internal/users/cache/db.go

```diff
@@ -88,35 +88,41 @@ type groupToUsersDB struct {
 }
 
 // New creates a new database cache by creating or opening the underlying db.
-func New(cacheDir string) (cache *Cache, err error) {
+func New(cacheDir string) (cache *Cache, rebuilt bool, err error) {
 	dbPath := filepath.Join(cacheDir, dbName)
 	defer decorate.OnError(&err, "could not create new database object at %q", dbPath)
 
 	var db *bbolt.DB
 	var i int
 	for {
-		db, err = openAndInitDB(dbPath, filepath.Join(cacheDir, dirtyFlagDbName))
+		var wasRebuilt bool
+
+		db, wasRebuilt, err = openAndInitDB(dbPath, filepath.Join(cacheDir, dirtyFlagDbName))
+		if wasRebuilt {
+			rebuilt = true
+		}
 		if err == nil {
 			break
 		}
 
 		if errors.Is(err, shouldRetryDBError{}) {
 			if i == 3 {
-				return nil, errors.Unwrap(err)
+				return nil, rebuilt, errors.Unwrap(err)
 			}
 			i++
 			continue
 		}
 
-		return nil, err
+		return nil, rebuilt, err
 	}
 
-	return &Cache{db: db, mu: sync.RWMutex{}, dirtyFlagPath: filepath.Join(cacheDir, dirtyFlagDbName)}, nil
+	return &Cache{db: db, mu: sync.RWMutex{}, dirtyFlagPath: filepath.Join(cacheDir, dirtyFlagDbName)}, rebuilt, nil
 }
 
 // openAndInitDB open a pre-existing database and potentially intializes its buckets.
 // It clears up any database previously marked as dirty or if it’s corrupted.
-func openAndInitDB(path, dirtyFlagPath string) (*bbolt.DB, error) {
+// If the database got cleared, it returns true as second argument.
+func openAndInitDB(path, dirtyFlagPath string) (*bbolt.DB, bool, error) {
 	if _, err := os.Stat(dirtyFlagPath); err == nil {
 		clearDatabase(path, dirtyFlagPath)
```

Comment:
As discussed. remove dirtyFlagPath from it and have the dirty flag only on manager, as it’s the one collecting the state now and issuing cleaning.